### DISCUSSION TITLE
Minimap

### DISCUSSION THREAD

**Discussion by mcp1766** [phcode-dev/phoenix#1197] [Q&A]
Hey,

I'am under Manjaro and use Chrome.  Trying to install minimap extension but fails.  Any idea if it is working or can we manually install it (and where) ? Thk

**Comment by abose** [comment]
Minimap extension should work in chrome. What is the error message?

**Reply by mcp1766** [reply]
Give some screenshots.  I followed this guide to install phcode:

> 

![Screenshot_20231122_175937](https://github.com/phcode-dev/phoenix/assets/27546954/5daffadd-1847-4bfc-8032-b6570ecceefe)

![Screenshot_20231122_180014](https://github.com/phcode-dev/phoenix/assets/27546954/9d10eaf1-c071-4e90-9c9f-495c6dc6c29f)

![Screenshot_20231122_180111](https://github.com/phcode-dev/phoenix/assets/27546954/ae13983c-d12a-4962-8309-fd5a33be1d71)

Sorry it is in french so am I.

**Reply by abose** [reply]
Press F12 and from console, Can you paste the error in devtools after installing the extension?

**Reply by mcp1766** [reply]
I'am not sure about what you want.  I tried lorem ipsum then wait until error then hit F12 then hard copy of terminal output.  If it is not correct can you give more details about the steps.. I'am just newbie.  Thk.
![Screenshot_20231122_191118](https://github.com/phcode-dev/phoenix/assets/27546954/a5616382-e62f-46af-a178-6a1c4b30a4c8)

![Screenshot_20231122_194913](https://github.com/phcode-dev/phoenix/assets/27546954/a22b8c23-ad2a-42ca-9d45-a7608a1da27d)

**Reply by abose** [reply]
Oh wait it looks like you built it from source.

There is a hosted version available at https://phcode.dev . that is the app to use unless you are developing Phoenix itself.

**Reply by mcp1766** [reply]
Just installed Chromium (from offical Manjaro repo).  I could install minimap and lorem ipsum extensions (what I could not with google Chrome).  But could not install emmet in Chromium.  I begin to believe that the problem is maybe with the browser somewhere...

**Reply by mcp1766** [reply]
I already used the hosted version.  Of course there I was able to install extensions but after restarting the browser I had to reinstall all the extensions and themes.  This hosted version seems not to save the user settings.  Never It asked for an account.  Do I am right ?

**Reply by abose** [reply]
Yes, all changes are saved to your browser and not uploaded anywhere. It was purposefully made account less for simple code editing that are in the local machine. we will add accounts only for cloud workflows. But even then an account less mode will be available.

Coming back to the actual problem, all symptoms point to browser storage being cleared across browser restarts. Chromium is a supported browser and infact we do all our tests in chromium.

**Reply by abose** [reply]
Let me try to repro this issue at my end. Can you givethe machine configuration?

Manjaro linux version:
Chromium install command or method used:
Network: wired/wireless, speed

**Reply by mcp1766** [reply]
I dont known how to avoid clearing storage with Chrome or chromium.  I send you some pictures about what happening. For infos:

> Manjaro kde plasma 5.27.9; kde framework 5.112.0, qt 5.15.11, kernel 6.5.11-1 Manjaro 64 bits, plateform graphique X11.  Google-chrome 119.0.6045.159 installed with yay -S google-chrome.   Connxion speel 400mbits (Voo.be) Ethernet link.
Chromium is 119.0.6045.159.  Installed from Start menu - System - Add/remove softaware.

Same issue with Chromium.

Some pictures
First start with defaults
![chrome_intial start](https://github.com/phcode-dev/phoenix/assets/27546954/9b9f98a9-7152-40df-a377-9b8d39f1dfaf)
Changing fonts isze and family
![chrome_phcode_change_font](https://github.com/phcode-dev/phoenix/assets/27546954/45f97d58-fcde-4f11-ad2f-5fd9187ff1cc)
Applying 
![chrome-font-applied](https://github.com/phcode-dev/phoenix/assets/27546954/2be7d332-59ba-4832-93df-b7c3d1931ee7)
Restarting Chrome - reset to default values
![Chrome-after-restart](https://github.com/phcode-dev/phoenix/assets/27546954/883393f2-2681-4602-a7e4-b0115766f353)
Try to set dark theme before restarting Chrome
![phcode-set-in-black-before-restart-chrome](https://github.com/phcode-dev/phoenix/assets/27546954/25829c6f-762c-4067-b3be-a2756deb1d4f)
Reset to white after relaunching Chrome and phcode
![phcode-reset-in-white-after-chrome-restart](https://github.com/phcode-dev/phoenix/assets/27546954/90d51503-7700-45eb-b99f-c7635e6ac27e)

How can I fix clearing storage in Chrome and Chromium ?

**Reply by abose** [reply]
To check if its a browser storage related issue, can you do the following:
1. Create a new file in the default project- (right click new file) ![image](https://github.com/phcode-dev/phoenix/assets/5336369/95f82343-a3e9-4664-9d1e-612fb276f3b5)
2. Create a `sample.html` file and write some contents.
3. After saving, close and relauch chrome. see if the file is present in the default project. If the file is present, then the browser storage is alright.

**Reply by mcp1766** [reply]
I did but gone after restart

**Reply by abose** [reply]
Looks like the browser is deleting all site data after each restart. This is not chrome's default behavior and some settings flag must be clearing it or the user profile folder used by chrome may have sone permission issues. That needs to but fixed in chrome and phcode should work after. 

https://unix.stackexchange.com/questions/587782/chrome-brave-keeps-signing-out-of-websites-on-x11

NB: A native linux version will be available by jan 2024, so browser issues like this will not be a problem there.

**Reply by mcp1766** [reply]
To end up I just made a fresh install of Manjaro and Chromium and Google Chrome (without any extensions) on another computer.  Google Chrome still not keeping settings, extensions, .. but Chromium Well (except for some extensions like minimap, lorem ipsum, ..).  Do a further test on a debian version.  Google Chrome and Chromium are ok to keep settings, themes, extensions (even for minimap, lorem ipsum,.). except for some other extensions like Brackets snippets for example.  So maybe is something to do with arch based os..   Last remarks, when creating a new file on a local folder the attributs are -rw--------username:username  which gives Apache server access denied (error 403).  Need  chmod og+rw to avoid this situation.  Thk for your good support.

**Reply by abose** [reply]
Thanks for the thorough investigations, it helps us identify issues in platforms that we dont test often. 

Tagging our resident Linux expert @charlypa to take a look.

**Reply by abose** [reply]
Browser issues like this might be hard for us to address just from phcode as websites typically cannot alter the browser settings. But its a genuine user issue and thats why we are working on native linux(+mac/win) builds to be in beta in a month or two. Much appreciate if you can take a look at our native builds later when its available @mcp1766 . Thanks again for the investigations done at your end.
